### ISSUE TITLE
Impossible to minimize MAE/MAPE using GPU

### ISSUE THREAD

**Issue by wutas** [catboost/catboost#1105]
GPU: TESLA K80

Hello, I can't fit the model using GPU by Google Colab. For exaple:
`cat = CatBoostRegressor(
        iterations=25000,
    loss_function='MAE',
    custom_metric=['MAPE','RMSE'],
    l2_leaf_reg=0.1,
    learning_rate=0.01,
    task_type='GPU',
    verbose = 100
)`

`cat.fit(
    X_train, y_train,
    cat_features=categorical_features_indices,
    eval_set=(X_val, y_val),
    early_stopping_rounds=3000
);`
 

The output is: 
0:	learn: 93783.0281697	test: 99906.7821997	best: 99906.7821997 (0)
100:	learn: 93783.0155843	test: 99906.7821997	best: 99906.7727924 (2)
200:	learn: 93783.0281697	test: 99906.7821997	best: 99906.7727924 (2)	
300:	learn: 93783.0218770	test: 99906.7916070	best: 99906.7727924 (2)	
400:	learn: 93783.0155843	test: 99906.7916070	best: 99906.7727924 (2)	

As we can see the model can't minimize MAE function. But if I use CPU:
`cat = CatBoostRegressor(
        iterations=25000,
    loss_function='MAE',
    custom_metric=['MAPE','RMSE'],
    l2_leaf_reg=0.1,
    learning_rate=0.01,
    verbose = 100
)`

I'll get output:
0:	learn: 93057.3786370	test: 99171.9336101	best: 99171.9336101 (0)	
100:	learn: 46468.8840847	test: 50816.4714564	best: 50816.4714564 (100)	
200:	learn: 27561.9279457	test: 29890.0993482	best: 29890.0993482 (200)	
300:	learn: 19882.6101584	test: 21100.3218046	best: 21100.3218046 (300)	
400:	learn: 16296.1568954	test: 16998.6450500	best: 16998.6450500 (400)	
500:	learn: 14540.2153033	test: 15295.1791825	best: 15295.1791825 (500)

So fitting goes OK.

P.S.
If I use RMSE loss_function on GPU  fitting goes OK too.

**Comment by LyzhinIvan** [comment]
Hi, @wutas ! Because of the nature of MAE loss function (gradient for each point equals to +1 or -1 and doesn't depend on scale of target values), fitting with gradient method is very poor. We have implemented Exact leaf estimation method for MAE loss function (also for Quantile and MAPE) for CPU that is much better and used by default. Currently it's not available for GPU. So I can recommend you to train your model with CPU, use RMSE with GPU, try to significantly increase learning rate or rescale your target values.

**Comment by guilhermesilva07113** [comment]
Hi!

I saw on [loss-functions](https://catboost.ai/en/docs/concepts/loss-functions-regression#usage-information) that `MAE` is `GPU` supported.

But when I tried to run  with the following hyperparameters:

```python
{
        "learning_rate": 0.2,
        "depth": 4,
        "l2_leaf_reg": 1,
        "random_state": 42,
        "verbose": False,
        "loss_function": "MAE",
        "task_type": "GPU",
    },
```
I received the following text:

```sh
Default metric period is 5 because MAE is/are not implemented for GPU
```

Why/when does it happen?
